File: AST.Server/Models/Message.cs
using System.ComponentModel.DataAnnotations;

namespace AST.Server.Models
{
    public class Message
    {
        public int Id { get; set; }
        [Required]
        public string Text { get; set; }
        public string SenderId { get; set; }
        public string ReceiverId { get; set; }
        public bool IsRead { get; set; }
        public int? ReplyToMessageId { get; set; }
        public List<string> Attachements { get; set; }
        public DateTime Timestamp { get; set; }

    }
}

File: ast.client/src/app/Models/Message.ts
export class Message {
  id!: number;
  content!: string;
  senderId!: string;
  receiverId!: string;
  senderName!: string;
  isRead!: boolean;
  replyToMessageId!: number;
  attachments!: string[];
  timestamp!: Date;
}
